packages/mobile/src/components/payment-method/PaymentMethod.tsx

```diff
@@ -72,12 +78,22 @@ export const PaymentMethod = ({
 }: PaymentMethodProps) => {
   const styles = useStyles()
   const neutral = useColor('neutral')
+  const dispatch = useDispatch()
 
   const balanceCents = formatUSDCWeiToFloorCentsNumber(
     (balance ?? new BN(0)) as BNUSDC
   )
   const balanceFormatted = formatCurrencyBalance(balanceCents / 100)
   const purchaseVendor = useSelector(getPurchaseVendor)
+  const { isEnabled: isCoinflowEnabled, isLoaded: isCoinflowEnabledLoaded } =
+    useFeatureFlag(FeatureFlags.BUY_WITH_COINFLOW)
+
+  // Initial state is coinflow by default, but set to stripe if coinflow is disabled.
+  useEffect(() => {
+    if (isCoinflowEnabledLoaded && !isCoinflowEnabled) {
```

Comment:
We should find a better pattern for feature flags that allows a single check on the value that waits until the config is loaded. It seems error-prone that we need to remember each time to check the loaded state before reading the value.

Reply:
yeah i agree that is exactly what happened here.